File: 807-Max_Increase_to_Keep_City_Skyline/code.cpp
#include <algorithm>
class Solution {
public:
    int maxIncreaseKeepingSkyline(vector<vector<int>>& grid) {
        int N = grid.size();
        int sum = 0;
        vector<int> Top(N);
        vector<int> Left(N);      
        int i,j;
        
        for(i=0; i<N; i++){
            for(j=0; j<N; j++){
                if (grid[i][j] > Left[i]){
                    Left[i] = grid[i][j];
                }
                if (grid[i][j] > Top[j]){
                    Top[j] =  grid[i][j];
                }
            }
        }
        
        for(i=0; i<N; i++){
            for(j=0; j<N; j++){
                sum += min(Left[i],Top[j]) - grid[i][j];
            }
        }
        
        return sum;
    }
};

File: 202-Happy_Number/code.py
class Solution:
    def isHappy(self, n):
        """
        :type n: int
        :rtype: bool
        """
        sum = 0
        sett = set()
        
        while n != 1:
            while n != 0:
                sum += (n % 10) **2
                n //= 10
            if sum  not in sett:
                sett.add(sum)
            else: return False
            n = sum
            sum = 0
        
        return True
        